kong/router.lua

```diff
@@ -768,49 +750,35 @@ function _M.new(routes)
 
           if matched_route then
             local upstream_host
-            local upstream_uri   = req_uri
+            local upstream_uri
             local upstream_url_t = matched_route.upstream_url_t
             local matches        = ctx.matches
 
-            -- URI stripping logic
 
-            local uri_root = req_uri == "/"
-
-            if not uri_root and matched_route.strip_uri
-               and matches.stripped_uri
-            then
-              upstream_uri = matches.stripped_uri
-            end
+            -- Path construction
 
-            -- uri trailing slash logic
+            -- if we do not have a path-match, then the postfix is simply the
+            -- incoming path, without the initial slash
+            local request_postfix = matches.stripped_uri or sub(req_uri, 2, -1)
```

Comment:
This variable should only be needed within the below `if matched_route.strip_uri` branch. Let's move it within it then. Doing so would also avoid executing `sub(req_uri, 2, -1)` twice.

Reply:
see comment below, fix it in after-access